lib/templates.js

```diff
@@ -344,41 +334,62 @@ const createEventResourceTemplate = (serviceName, serviceDefinition, appConfig,
         visibility,
         resourceDefinitions: [_getResourceDefinition("asyncapi-v2", "json", ordId, serviceName, "asyncapi2.json", accessStrategies)],
         extensible: { supported: "no" },
-        entityTypeMappings: (appConfig.entityTypeTargets?.length > 0)
-                ? [{
-                    entityTypeTargets: appConfig.entityTypeTargets.map(m => ({ ordId: m.ordId }))
-                }]
-                : (Array.isArray(appConfig.entityTypeMappings)
-                    ? [{
-                        entityTypeTargets: _.uniqBy(
-                            appConfig.entityTypeMappings.flatMap(m => m.entityTypeTargets).map(et => ({ ordId: et.ordId })),
-                            'ordId'
-                        )
-                    }]
-                    : []),
+        entityTypeMappings: _getEntityTypeMappings(serviceDefinition),
         ...ordExtensions
     };
 
     return obj.visibility === RESOURCE_VISIBILITY.public || obj.visibility === RESOURCE_VISIBILITY.internal ? [obj] : [];
 };
 
+function _getEntityTypeMappings(serviceDefinition) {
+    const entities = Object.values(serviceDefinition.entities).flatMap(
+        entity => _assertionsFlatList(entity)
+    ).map(createEntityTypeMappingsItemTemplate)
+    const entityTypeTargets = _.uniqBy(entities, CONTENT_MERGE_KEY).filter(entity => entity !== undefined).map(({ordId}) => ({
+        ordId
+    }))
+    if (entityTypeTargets.length > 0) {
+        return [{entityTypeTargets}]
+    } else {
+        return []
+    }
+}
+
+function _assertionsFlatList(currentEntity, processedEntities = []) {
+    if(!currentEntity.associations) {
+        return [currentEntity]
+    }
+    const entityAssociationTargets = Object.values(currentEntity.associations).map(association => ({target: association.target, entity: association._target}))
+
+    const assertionsTodo = []
+    entityAssociationTargets.forEach(({target, entity}) => {
+        if (processedEntities.includes(target)) {
+            return;
+        }
+        processedEntities = [...processedEntities, target]
+        assertionsTodo.push(entity)
+    })
+
+    return [currentEntity, ...assertionsTodo.flatMap(entity => _assertionsFlatList(entity, processedEntities))]
+}
+
```

Comment:
```suggestion
function _flattenEntityGraph(currentEntity, processedEntities = []) {
    if(!currentEntity.associations) {
        return [currentEntity]
    }
    const entityAssociationTargets = Object.values(currentEntity.associations).map(association => ({target: association.target, entity: association._target}))

    const entityToProcess = []
    entityAssociationTargets.forEach(({target, entity}) => {
        if (processedEntities.includes(target)) {
            return;
        }
        processedEntities = [...processedEntities, target]
        entityToProcess.push(entity)
    })

    return [currentEntity, ...entityToProcess.flatMap(entity => _flattenEntityGraph(entity, processedEntities))]
}

```
Implementation is good. I would suggest that names the variables more clear.